packages/core/lib/commands/test/run.js

```diff
@@ -107,12 +107,30 @@ module.exports = async function (options) {
     }
   };
 
+  function sanitizeNetworkID(network_id) {
+    if (network_id !== "*") {
+      if (!parseInt(network_id, 10)) {
+        const error =
+          `The network id specified in the truffle config ` +
+          `(${network_id}) is not valid. Please properly configure the network id as an integer value.`;
+        throw new Error(error);
+      }
+      return network_id;
+    } else {
+      // We have a "*" network. Return the default.
+      return 4447;
+    }
+  }
```

Comment:
A couple of notes on [`parseInt`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt). 

- It could return `0`, or `NaN` both of which are falsey. If you intended this behavior, it warrants a comment.  
- has a [peculiar behavior which allows `parseInt("23*blahasdf#$", 10) === 23`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt#using_parseint).  The question is how strict do you want to be?

Reply:
> This logic could be simplified if checking for `*` first.
> 
> ```js
> // use default network_id if *
> if (network_id === "*") return 4447;
> 
> const parsedNetworkId = parseInt(network_id, 10);
> if (isNaN(parsedNetworkId)) { //throw appropriate error }
> return parsedNetworkId;
> ```

Thanks for the suggestion @cds-amal . I will simplify the function as you suggested. It is the same function that is written for truffle develop. So, once it is simplifed, I think we can simplify the sanitize function there too.

Reply:
> A couple of notes on [`parseInt`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt).
> 
> * It could return `0`, or `NaN` both of which are falsey. If you intended this behavior, it warrants a comment.
> * has a [peculiar behavior which allows `parseInt("23*blahasdf#$", 10) === 23`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt#using_parseint).  The question is how strict do you want to be?

I am not that sure how far should we validate this to check for junk values. Any suggestions @eggplantzzz ?

Reply:
I like @cds-amal's suggestion for re-writing this. I think we probably should do that and implement it with `truffle develop` as well. As for how far we should go with this, I'm not too fired up about going hard on this. I think what Amal has suggested should be good enough for now in my opinion.

Reply:
I agree with you @eggplantzzz . Lets modify this in this way.